include/nbl/builtin/hlsl/scene/keyframe.hlsl

```diff
@@ -6,49 +6,80 @@
 #ifndef _NBL_BUILTIN_HLSL_SCENE_KEYFRAME_INCLUDED_
 #define _NBL_BUILTIN_HLSL_SCENE_KEYFRAME_INCLUDED_
 
-
-
 #include <nbl/builtin/hlsl/math/quaternions.hlsl>
 
-
-
 namespace nbl
 {
 namespace hlsl
 {
 namespace scene
 {
-	using namespace math;
 
+using namespace math;
+
+struct Keyframe_t
+{
+	uint2 data[3];
 
-	struct Keyframe_t
+	float3 getScale()
 	{
-		uint2 data[3];
+		return decodeRGB18E7S3(data[2]);
+	}
 
-		float3 getScale();
-		quaternion_t getRotation();
-		float3 getTranslation();
-	};
+	quaternion_t getRotation()
+	{
+		return { decode8888Quaternion(data[1][1]) };
+	}
 
+	float3 getTranslation()
+	{
+		return uintBitsToFloat(uint3(data[0].xy, data[1][0]));

```

Comment:
find out what the HLSL equivalent of GLSL's `uintBitsToFloat` is

Reply:
`asfloat`